test-data/unit/check-warnings.test

```diff
@@ -145,8 +145,29 @@ main:4: warning: Returning Any from function declared to return "int"
 
 [case testReturnAnyForNotImplementedInBinaryMagicMethods]
 # flags: --warn-return-any
+from typing import Any
 class A:
-    def __eq__(self, other: object) -> bool: return NotImplemented
+    def __init__(self, x: Any) -> None:
+      self.x = x
+    def __eq__(self, other: Any) -> bool:
+      if self.__class__ is other.__class__:
+        return bool(self.x == other.x)
+      return NotImplemented
+    def __lt__(self, other: Any) -> bool: return NotImplemented
+    def __gt__(self, other: Any) -> bool: return NotImplemented
+    def __le__(self, other: Any) -> bool: return NotImplemented
+    def __ge__(self, other: Any) -> bool: return NotImplemented
+    def __ne__(self, other: Any) -> bool:
+      if self.__class__ is other.__class__:
+        return bool(self.x != other.x)
+      else:
+        # pylint will complain about this return not needing
```

Comment:
This comment seems irrelevant; we don't run pylint on test cases.

Reply:
The reason I put this comment in is that someone might think that the "else" could be removed, leaving the "return" outside then if-then-else (pylint encourages this kind of thing). Do you want me to just remove the comment about pylint complaining?

Reply:
I don't think the comment is helpful, but I won't be the person deciding whether to merge this.

Why does it matter whether the `else` is there? It seems to me that it shouldn't matter to mypy, since the control flow should be the same in both cases.

Reply:
The type inferencing might produce different results (I'm essentially doing black-box test thinking). So, I have two ways of generating a Union result.